### PR TITLE
Implemented cluster checks to warn for tables that need to be recreated or upgraded

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: blackbox/docs/sql/system.txt [hunk 1]
```diff
@@ -1132,13 +1132,14 @@ Example query::
   +----+--------------------------------------------------------------...-+
   |  1 | The setting 'discovery.zen.minimum_master_nodes' must not be ... |
   |  2 | The total number of partitions of one or more partitioned tab... |
+  |  3 | The following tables must be recreated for compatibility with... |
```
Comment (dobe) [inline blackbox/docs/sql/system.txt]: woud rather say "need to"

File: sql/src/main/java/io/crate/operation/collect/sources/SystemCollectSource.java [hunk 2]
```diff
@@ -95,38 +97,62 @@ public SystemCollectSource(ClusterService clusterService,
...
-            .put(SysOperationsLogTableInfo.IDENT.fqn(), jobsLogs::operationsLogGetter)
-            .put(SysChecksTableInfo.IDENT.fqn(), new SysChecker<>(sysChecks)::checksGetter)
-            .put(SysNodeChecksTableInfo.IDENT.fqn(), () -> sysNodeChecks)
-            .put(SysRepositoriesTableInfo.IDENT.fqn(), sysRepositoriesService::repositoriesGetter)
-            .put(SysSnapshotsTableInfo.IDENT.fqn(), sysSnapshots::snapshotsGetter)
-            .put(SysSummitsTableInfo.IDENT.fqn(), new SummitsIterable()::summitsGetter)
-            .put(PgTypeTable.IDENT.fqn(), pgCatalogTables::typesGetter)
+        iterableGetters = ImmutableMap.<String, Supplier<CompletableFuture<? extends Iterable<?>>>>builder()
+            .put(InformationSchemataTableInfo.IDENT.fqn(),
+                () -> CompletableFuture.completedFuture(informationSchemaIterables.schemas()))
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/collect/sources/SystemCollectSource.java]: would probably make sense to extract a method like synchronousGetter(...)

File: sql/src/main/java/io/crate/operation/collect/sources/SystemCollectSource.java [hunk 3]
```diff
@@ -135,11 +161,15 @@ public SystemCollectSource(ClusterService clusterService,
...
-        Supplier<Iterable<?>> iterableGetter = iterableGetters.get(table);
+        Supplier<CompletableFuture<? extends Iterable<?>>> iterableGetter = iterableGetters.get(table);
         assert iterableGetter != null : "iterableGetter for " + table + " must exist";
-        return ImmutableList.<CrateCollector>of(
-            new RowsCollector(downstream, toRowsIterable(collectPhase, iterableGetter.get(),
-                downstream.requirements().contains(Requirement.REPEAT))));
+        return ImmutableList.of(
+            new RowsCollector(
+                downstream,
+                iterableGetter.get(),
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/collect/sources/SystemCollectSource.java]: ftr: pass the getter - do not call it

File: sql/src/main/java/io/crate/operation/reference/sys/check/cluster/TablesNeedUpgradeSysCheck.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    @Override
+    public CompletableFuture<?> computeResult() {
+        Collection<String> tablesNeedRecreation =
+            LuceneVersionChecks.tablesNeedRecreation(clusterService.state().metaData());
+
+        final CompletableFuture<Collection<String>> result = new CompletableFuture<>();
+        try {
+            Object[] sqlParam = tablesNeedRecreation.toArray(new String[]{});
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/reference/sys/check/cluster/TablesNeedUpgradeSysCheck.java]: would it make sense to somehow share the impl with AlterTableOperation.executeAlterTableAddColumn

**Comment by matriv** [comment]
retest this please

File: sql/src/main/java/io/crate/executor/ExecuteSQL.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+package io.crate.executor;
+
+import io.crate.action.sql.Option;
+import io.crate.action.sql.ResultReceiver;
+import io.crate.action.sql.SQLOperations;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ExecuteSQL {
```
Comment (matriv) [inline sql/src/main/java/io/crate/executor/ExecuteSQL.java]: I removed it...

**Pull request merged**